spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -193,15 +200,41 @@ public boolean supportColumnarReads(InputPartition partition) {
     }
   }
 
+  static long numFilesToScan(CombinedScanTask scanTask) {
+    long fileCount = 0L;
+    for (FileScanTask file : scanTask.files()) {
+      fileCount += 1L;
```

Comment:
should we incorporate delete's as well assosiated with the FileScanTask
```suggestion
      fileCount += 1L + file.deletes().size();
```

Reply:
We could have a separate metric for number of delete files to read. Or rather, a sum of the number of delete files to apply, over the data files to be read; we won't actually track what the delete files are, if there is any overlap among them (can there be overlap?).
@rdblue @aokolnychyi what do you think?

Reply:
I think it's fair to count all the delete files, but definitely in another metric. 

There will be duplicates in two cases
1. It's an equality delete
2. There is a position delete file for a data file which is split into being read in two different tasks (this is also a problem for duplicate files being read)